**Two fixes required before this can merge.**

**1. TypeScript error — `DownloadTask` type does not have `url` or `destination` properties**

In `getActiveSession`, the reconnection logic reads `task.url` and `task.destination` from the result of `getExistingDownloadTasks()`. The library's `DownloadTask` type does not expose these properties, hence the two TS errors:

```
error TS2339: Property 'url' does not exist on type 'DownloadTask'
error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'
```

Session reconnection after app restart is deferred to a future version (it is listed as a deferred item in the design doc). Remove `getActiveSession` entirely — it should not be implemented in this task. The method is not part of `IDownloadManager` and is not called by anything yet.

**2. ESLint error — unused `expectedBytes` parameter**

In the `begin` callback, `expectedBytes` is destructured but never used. Per the project lint rules, unused args must be prefixed with `_`. Fix:

```typescript
this.task.begin(({ expectedBytes: _expectedBytes }: { expectedBytes: number }) => {
```

Or simply use an empty destructure since the value is not needed at all:

```typescript
this.task.begin((_: unknown) => {
```

No other changes needed. The rest of the implementation is correct — `start()`, `stop()`, `progress` mapping, `done` with `video:///` prefix, `factory.ts` wiring, and the `pct` format (note: the task specified `toFixed(0)` to match desktop format, but `toFixed(1)` is also acceptable unless the DownloadModal UI displays it raw — not a blocking issue, leave as-is).

--- a/src/bindings/download/index.ts
+++ b/src/bindings/download/index.ts
@@ -1,6 +1,6 @@
 import RNFS from 'react-native-fs';
 import { EventEmitter } from 'events';
-import { createDownloadTask, getExistingDownloadTasks } from '@kesha-antonov/react-native-background-downloader';
+import { createDownloadTask } from '@kesha-antonov/react-native-background-downloader';
 import path from 'path-browserify';
 import { IDownloadManager, IDownloadSession, DownloadProps } from 'incyclist-services';
 
@@ -56,7 +56,7 @@ export class MobileDownloadSession extends EventEmitter implements IDownloadSess
             return;
         }
 
-        this.task.begin(({ expectedBytes }: { expectedBytes: number }) => {
+        this.task.begin(({ expectedBytes: _expectedBytes }: { expectedBytes: number }) => {
             if (this.stopped) {
                 return;
             }
@@ -104,24 +104,6 @@ export class MobileDownloadSession extends EventEmitter implements IDownloadSess
 export class MobileDownloadManager implements IDownloadManager {
     public createSession(url: string, fileName: string, _props?: DownloadProps): IDownloadSession {
         return new MobileDownloadSession(url, fileName);
-    }
-
-    /**
-     * Reconnects to an existing background task after app restart.
-     */
-    public async getActiveSession(sessionId: string): Promise<IDownloadSession | null> {
-        const tasks = await getExistingDownloadTasks();
-        const task = tasks.find(t => t.id === sessionId);
-        
-        if (task) {
-            const session = new MobileDownloadSession(task.url, task.destination);
-            // Inject existing task and attach handlers
-            (session as any).task = task;
-            (session as any).attachHandlers();
-            return session;
-        }
-        
-        return null;
     }
 
     /**

